writer_test.go

```diff
@@ -565,3 +571,85 @@ func TestBloomFilterForDict(t *testing.T) {
 		t.Error("bloom filter should have contained 'test'")
 	}
 }
+
+func TestEmptyFile(t *testing.T) {
+	f := func(t *testing.T, schema *parquet.Schema) {
+		tmp, err := os.CreateTemp("/tmp", "*.parquet")
+		if err != nil {
+			t.Error(err)
+		}
+		defer tmp.Close()
+		path := tmp.Name()
+		defer os.Remove(path)
+
+		writer := parquet.NewWriter(tmp, schema)
+		if err := writer.Close(); err != nil {
+			t.Error(err)
+		}
+
+		fs, err := tmp.Stat()
+		if err != nil {
+			t.Error(err)
+		}
+
+		_, err = parquet.OpenFile(tmp, fs.Size())
+		if err != nil && !errors.Is(err, parquet.ErrMissingRootColumn) {
+			t.Errorf("expected error %s, got %s", parquet.ErrMissingRootColumn, err)
+		}
+	}
+
+	tests := []struct {
+		scenario string
+		schema   *parquet.Schema
+	}{
+		{
+			scenario: "empty file with no schema",
+			schema:   nil,
+		},
+		{
+			scenario: "empty file with schema",
+			schema:   parquet.SchemaOf(new(firstAndLastName)),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.scenario, func(t *testing.T) {
+			f(t, test.schema)
+		})
+	}
+}
+
+func TestFileWithNoSchema(t *testing.T) {
```

Comment:
Wait... the PR says we _shouldn't_ panic here?

Reply:
fixed